### PR TITLE
Make run_tests.py assert that the ios test dylib is at least as new as libFlutter.dylib

### PR DESCRIPTION
**Pull request by dnfield**
This would have helped me figure out why I wasn't getting the same failures as on CI more quickly.

### CODE REVIEW

**Review by jmagman — COMMENTED** [review]
What if there were no changes to the libFlutter but I'm iterating on the tests themselves?

**Comment by dnfield** [comment]
> What if there were no changes to the libFlutter but I'm iterating on the tests themselves?

Then the test dylib will be newer and this will pass.

File: testing/run_tests.py [hunk 1]
```diff
@@ -629,6 +629,15 @@ def ensure_ios_tests_are_built(ios_out_dir):
   assert os.path.exists(tmp_out_dir
                        ) and os.path.exists(ios_test_lib), final_message
 
+  ios_test_lib_time = os.path.getmtime(ios_test_lib)
+  flutter_dylib = os.path.join(tmp_out_dir, 'libFlutter.dylib')
+  flutter_dylib_time = os.path.getmtime(flutter_dylib)
+
+  final_message = '%s is older than %s. Please run the following commands: \n%s' % (
```
Comment (zanderso) [inline testing/run_tests.py:636]: `autoninja` is deprecated. Could you also change line 625 to be `'ninja -C ...'` instead?

**Review by zanderso — APPROVED** [review]

File: testing/run_tests.py (hunk 1 above)
Comment (dnfield) [reply testing/run_tests.py:636]: done

**Pull request merged**